File: scratch/check_db.py
import sqlite3
import json

conn = sqlite3.connect('pb_data/data.db')
cursor = conn.cursor()

# Find transaction with number like %451%
cursor.execute("SELECT id, number, description, date, third_party_id, tx_type_id FROM transactions WHERE number LIKE '%451%'")
tx_rows = cursor.fetchall()
print("=== TRANSACTIONS matching '%451%' ===")
for r in tx_rows:
    print(r)
    tx_id = r[0]
    
    # Check invoices for this tx_id
    cursor.execute("SELECT id, number, notes, cross_doc_ref, subtotal, iva_total, total FROM invoices WHERE tx_id = ?", (tx_id,))
    inv_rows = cursor.fetchall()
    print(f"  Invoices for tx_id {tx_id}:")
    for inv in inv_rows:
        print("   ", inv)

    # Check tx_lines for this tx_id
    cursor.execute("SELECT id, description, cross_doc_ref, debit, credit FROM tx_lines WHERE tx_id = ?", (tx_id,))
    lines_rows = cursor.fetchall()
    print(f"  Tx Lines for tx_id {tx_id}:")
    for l in lines_rows:
        print("   ", l)

conn.close()

File: scratch/check_db.js
const sqlite3 = require('sqlite3').verbose();
const path = require('path');

const dbPath = path.resolve(__dirname, '../pb_data/data.db');
const db = new sqlite3.Database(dbPath, sqlite3.OPEN_READONLY);

db.serialize(() => {
  // Check if RC-00000595 exists
  db.get('SELECT * FROM transactions WHERE number = "RC-00000595"', [], (err, row) => {
    console.log('\n--- RC-00000595 ---');
    console.log(err || row || 'Not found');
  });

  // Check if RC-00000596 exists
  db.get('SELECT * FROM transactions WHERE number = "RC-00000596"', [], (err, row) => {
    console.log('\n--- RC-00000596 ---');
    console.log(err || row || 'Not found');
  });

  // Check max number
  db.get('SELECT max(number) as max_num FROM transactions WHERE number LIKE "RC-%"', [], (err, row) => {
    console.log('\n--- MAX RC NUMBER ---');
    console.log(err || row);
  });

  // Check table schema for transactions
  db.all('PRAGMA table_info(transactions)', [], (err, rows) => {
    console.log('\n--- TRANSACTIONS SCHEMA ---');
    console.log(err || rows);
    db.close();
  });
});
